File: scratch/get_remote_logs.js
const { Client } = require('ssh2');
const fs = require('fs');
require('dotenv').config({ path: './server/.env' });

const conn = new Client();
conn.on('ready', () => {
  console.log('Client :: ready');
  conn.exec('pm2 logs smartnanum-server --lines 20 --nostream', (err, stream) => {
    if (err) throw err;
    let data = '';
    stream.on('data', (chunk) => {
      data += chunk;
    }).on('close', (code, signal) => {
      fs.writeFileSync('scratch/remote_logs.txt', data);
      console.log('Logs saved to scratch/remote_logs.txt');
      conn.end();
    });
  });
}).connect({
  host: '210.114.22.136',
  port: 22,
  username: 'root',
  password: 'Ch070809'
});

File: scratch/get_remote_logs.py
import paramiko
import os
from dotenv import load_dotenv

load_dotenv('server/.env')

def get_logs():
    try:
        ssh = paramiko.SSHClient()
        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
        ssh.connect('210.114.22.136', username='root', password='Ch070809')
        
        stdin, stdout, stderr = ssh.exec_command('pm2 logs smartnanum-server --lines 50 --nostream')
        data = stdout.read()
        
        with open('scratch/remote_logs.txt', 'wb') as f:
            f.write(data)
            
        print("Logs saved successfully!")
        ssh.close()
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    get_logs()
